hail/src/main/scala/is/hail/expr/ir/lowering/LoweringPipeline.scala

```diff
@@ -1,29 +1,24 @@
 package is.hail.expr.ir.lowering
 
-import is.hail.expr.ir.{BaseIR, ExecuteContext, Optimize}
-import is.hail.utils.HailException
+import is.hail.expr.ir.{BaseIR, ExecuteContext}
+import is.hail.utils._
 
-case class LoweringPipeline(lowerings: IndexedSeq[LoweringPass]) {
+case class LoweringPipeline(lowerings: LoweringPass*) {
   assert(lowerings.nonEmpty)
   lowerings.zip(lowerings.tail).foreach { case (l, r) =>
     assert(l.after == r.before)
   }
 
-  final def apply(ctx: ExecuteContext, ir: BaseIR, optimize: Boolean): BaseIR = {
+  final def apply(ctx: ExecuteContext, ir: BaseIR): BaseIR = {
     var x = ir
 
-    if (optimize)
-      x = Optimize(x, noisy = true, context = "Lowerer, initial IR", ctx)
-
     lowerings.foreach { l =>
       try {
         x = l.apply(ctx, x)
-        if (optimize)
-          x = Optimize(x, noisy = true, context = s"${l.context}, post Lowering", ctx)
       } catch {
-        case e: HailException => throw e
         case e: Throwable =>
-          throw new RuntimeException(s"error while applying lowering '${ l.context }'", e)
+          log.error(s"error while applying lowering '${ l.context }'")
```

Comment:
slight change with what's going on in error handling -- I think this is better, in that it doesn't wrap errors and confuse people, but still puts the info in the log.

Reply:
This is an improvement for the user experience, but makes the dev experience worse. Right now, the wrapped stack trace (the one I actually care about) shows up in the IDE, with clickable line numbers. With the change, I would have to open the log, and navigate to the referenced lines manually.

Could we maybe control this behavior with an environment flag? Or is there a trick I don't know for working with stack traces in logs?

Reply:
the wrapped error is the only error now. I basically never used the stack trace to this line, because it's control flow code and the same every time.

Reply:
Yeah, just realized that. Never mind!